frontend/services/opengl_glfw/OpenGL_GLFW_Service.cpp

```diff
@@ -797,6 +801,10 @@ void OpenGL_GLFW_Service::postGraphRender() {
 
     ::glfwSwapBuffers(m_pimpl->glfwContextWindowPtr);
 
+#ifdef PROFILING
+    core::CallProfiling::CollectGPUPerformance();
```

Comment:
I would really like to avoid the GL Service knowing about calls and such. I think the `CollectGPUPerformance()` hints that we want a service to do the performance collection after the buffer swap?